feat(ui): Send to fossology error message.

smrutis1 opened the pull request:
* Added a dialog with message "Fossology is not configured" instead of the confusing message "Error" on clicking of send to fossology icon.
 * Added a switch in sw360.properties to enable or disable the the fossology connection.
 * Displayed the enabled/disabled  status in fossology admin portlet.

**Steps to Test:**

- Go to Projects-> Click on the "Send to Fossology" icon.
- Go to Components -> Release Overview -> Click on the "Send to Fossology" icon.
- Go to Projects -> Clearing Status -> Click on the "Send to Fossology" icon.
- Go to Components -> Clearing Details -> Click on the "Send to Fossology" icon.
In all the above cases, if the fossology connection is disabled, the user gets a dialog message that fossology is not configured.

Go to Admin-> Fossology -> A additional message should appear displaying the fossology connection status(i.e enabled/disabled).

Closes #494 

Signed-off-by: Smruti Sahoo <smruti.sahoo@siemens.com>

imaykay commented:
@smrutis1 @mcjaeger two remarks on this PR from my side:
1. Me as a user, I would prefer not being able to click the icon at all - instead of having to click it to see that it does nothing. A tooltip on the disabled (maybe gray icon or i-icon instead of the fossology icon) could tell the user that fossy is currently not configured).
2. I am not sure if another property then the connection configuration properties makes sense. Normally I would assume that the configuration is there if the configuration is there ;-) And not if another property tells me that the configuration should be there.
I am totally aware that the connection configuration is in the fossology service so there might be the need to query that one. But the response might be cached for 15mins or so, because the backend config will not be changed that often.

smrutis1 commented:
@imaykay , Thanks for the suggestion. I have added the disabled fossology grey icon with the tooltip instead of a dialogue and added a optional property in sw360.properties, by unblocking the same, we can disable the fossology connection after explicitly checking the same.

mcjaeger commented:
(from discussion) maybe calling the check connection on the FOSSology Admin page could enable the button of "Send to FOSSology" again without the need to restart the entire application.

nutanv1contr commented:
Hi @smrutis1 / @mcjaeger ,

Functionality is  working as expected except one minor issue.
**Issue** : In admin screen when user "check connectivity to fossology server" then if the status is changed from disabled to enables or vice verse then in that case the status does not reflect immediately.User needs to refresh the page to view the updated status.
**Scenarios tested:**:
1.Go to Projects-> Click on the "Send to Fossology" icon.
2.Go to Components -> Release Overview -> Click on the "Send to Fossology" icon.
3.Go to Projects -> Clearing Status -> Click on the "Send to Fossology" icon.
4.Go to Release -> Clearing Details -> Click on the "Send to Fossology" icon.
**Output:**
If the fossology connection does not exist then "Send to Fossology" icon is disabled with a tooltip "Connection is not configured"

5.Go to Admin>>Fossology and click on "Check connectivity to Fossology server"
**Output**
Fossology connection status gets updated to "Enabled/Disables" on refreshing the page.

smrutis1 commented:
@nutanv1contr  Please consider retesting for the reported issue.

nutanv1contr commented:
> @nutanv1contr Please consider retesting for the reported issue.

Issue is now resolved.

imaykay commented on frontend/sw360-portlet/src/main/webapp/html/admin/fossyAdmin/view.jsp:
```diff
@@ -77,6 +91,11 @@
             url: url,
             success: function (data) {
                 $resultElement.text(data.result);
+                if(data.result == 'SUCCESS'){
```
I think we might have a problem here: Since you change the fossology state in this generic ajax method, the state will be also be changed (only in client view here, not on server) when a script has been deployed successfully, which is the other operation on this page that uses this ajax method.

smrutis1 replied:
Thanks for pointing it out. Have corrected this one.

imaykay commented on frontend/sw360-portlet/src/main/webapp/html/admin/fossyAdmin/view.jsp:
```diff
@@ -57,9 +58,22 @@
 <button class="addButton" onclick="deployScripts('deployResult')">Deploy Scripts To Fossology Server</button>
 <span id="deployResult"></span>
 
+<div>Fossology Connection Status: <span id="conStatus"><b>${currentConnStatus}</b></span></div>
```
I think this page has not been a beauty before, but wouldn't it be better to place button and this text together (maybe in own block with heading)? And I am not sure if we need the existing output next to the button that pops up e.g. with "Failure" if the connection is not available.

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/FossologyConnectionHelper.java:
```diff
@@ -0,0 +1,45 @@
+
+	private static final Logger log = Logger.getLogger(FossologyConnectionHelper.class);
+
+	public static boolean FOSSOLOGY_CONNECTION_ENABLED;
```
Certainly, it was making easy to read the states in jsp. :)

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/FossologyConnectionHelper.java:
```diff
@@ -0,0 +1,45 @@
+			checkConnection = new ThriftClients().makeFossologyClient().checkConnection();
+		} catch (TException e) {
+			FOSSOLOGY_CONNECTION_ENABLED = false;
+			e.printStackTrace();
```
Removed it.

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/common/FossologyConnectionHelper.java:
```diff
@@ -0,0 +1,45 @@
+			e.printStackTrace();
+			log.error("Error connecting to backend", e);
+		}
+		FOSSOLOGY_CONNECTION_ENABLED = checkConnection.equals(RequestStatus.SUCCESS) ? true : false;
```
Removed it :-)

smrutis1 commented on frontend/sw360-portlet/src/main/java/org/eclipse/sw360/portal/hooks/FossologyCheckConnectionOnStartupHook.java:
```diff
@@ -0,0 +1,44 @@
+
+	@Override
+	public void run(HttpServletRequest request, HttpServletResponse response) throws ActionException {
+		if(!calledOnServerstartUp) {
```
This hook is executed for every portal request.

nutanv1contr commented:
Rested the functionality.It is working as expected after merge conflict resolve.
